### PR TITLE
Use pvxslibs instead of qsrv

### PR DESCRIPTION
**Pull request by coretl**
This is not currently working

### CODE REVIEW

**Comment by coretl** [comment]
@mdavidsaver I'm trying to use QSRV2 in pythonSoftIOC, but I'm missing something. When I make the changes in this PR, install into a venv, and run `venv/bin/python docs/examples/example_asyncio_ioc.py` I get:
```
Unable to find symbol 'pvar_dset_devLoPDBQ2UTag' : /venv/lib/python3.10/site-packages/epicscorelibs/lib/libdbCore.so.7.0.7.99.0: undefined symbol: pvar_dset_devLoPDBQ2UTag
Unable to find dset for record type 'longin' support 'devLoPDBQ2UTag' : (null)
Traceback (most recent call last):
  File "/pythonSoftIOC/docs/examples/example_asyncio_ioc.py", line 2, in <module>
    from softioc import softioc, builder, asyncio_dispatcher
  File "/venv/lib/python3.10/site-packages/softioc/__init__.py", line 28, in <module>
    raise RuntimeError('Error registering')
RuntimeError: Error registering
```
I thought these symbols might have been from the demo application, so commented out everything apart from the `registrar` calls in `pvxsIoc.dbd`, but then I get:
```
Unable to find symbol 'pvar_func_pvxsBaseRegistrar' : /venv/lib/python3.10/site-packages/epicscorelibs/lib/libdbCore.so.7.0.7.99.0: undefined symbol: pvar_func_pvxsBaseRegistrar
Unable to find registrar 'pvxsBaseRegistrar' : (null)
Traceback (most recent call last):
  File "/pythonSoftIOC/docs/examples/example_asyncio_ioc.py", line 2, in <module>
    from softioc import softioc, builder, asyncio_dispatcher
  File "/venv/lib/python3.10/site-packages/softioc/__init__.py", line 28, in <module>
    raise RuntimeError('Error registering')
RuntimeError: Error registering
```
I see these same errors when importing in CI:
https://github.com/dls-controls/pythonSoftIOC/actions/runs/5189205532/jobs/9353980317?pr=132

I am using pvxslibs 1.2.1a1.

I added the DSOs and include path in `setup.py`, then loaded the dbd in `__init__.py`, is there anything I missed?

File: softioc/builder.py [hunk 1]
```diff
@@ -261,20 +261,23 @@ def _long_string(fields):
...
-    fields['_wf_dtype'] = numpy.dtype('uint8')
+    fields['_wf_dtype'] = numpy.dtype('int8')
 
     fields['NELM'] = length
-    fields['FTVL'] = 'UCHAR'
+    fields['FTVL'] = 'CHAR'
 
+def qform_string(rec):
+    rec.add_info("Q:form", "String")
+    return rec
```
Comment (coretl) [inline softioc/builder.py:271]: This makes QSRV2 treat a waveform as a string...

File: softioc/builder.py [hunk 2]
```diff
@@ -261,20 +261,23 @@ def _long_string(fields):
 
     fields.setdefault('initial_value', '')
     fields['_wf_nelm'] = length
-    fields['_wf_dtype'] = numpy.dtype('uint8')
+    fields['_wf_dtype'] = numpy.dtype('int8')
 
     fields['NELM'] = length
-    fields['FTVL'] = 'UCHAR'
+    fields['FTVL'] = 'CHAR'
```
Comment (coretl) [inline softioc/builder.py:267]: ... but only if FTVL is CHAR

File: softioc/__init__.py [hunk 3]
```diff
@@ -17,11 +20,14 @@
...
-for dbd in ('base.dbd', 'PVAServerRegister.dbd', 'qsrv.dbd'):
-    dbLoadDatabase(dbd, os.path.join(path.base_path, 'dbd'), None)
+base_dbd_path = os.path.join(epicscorelibs.path.base_path, 'dbd')
+dbLoadDatabase('base.dbd', base_dbd_path, None)
+dbLoadDatabase('pvxsIoc.dbd', pvxslibs.path.dbd_path, None)
 iocStats = os.path.join(os.path.dirname(__file__), "iocStats", "devIocStats")
 dbLoadDatabase('devIocStats.dbd', iocStats, None)
 
+ctypes.CDLL(find_dso('pvxslibs.lib.pvxsIoc'), ctypes.RTLD_GLOBAL)
+os.environ.setdefault('PVXS_QSRV_ENABLE', 'YES')
```
Comment (coretl) [inline softioc/__init__.py:30]: pvxslibs needs to be loaded here, rather than linked in like dbCore, etc

**Review by AlexanderWells-diamond — APPROVED** [review]
Looks good to me. At some point we may want to add more tests for various other QSRV features but this is fine for now.

**Pull request merged**